src/precice/SolverInterface.hpp

```diff
@@ -774,6 +774,75 @@ class SolverInterface {
 
   ///@}
 
+  /** @name Experimental Data Access
+   * These API functions are \b experimental and may change in future versions.
+   */
+  ///@{
+
+  /**
+   * @brief setBoundingBoxes Define a region of interest on a received mesh
+   *        (<use-mesh ... from="otherParticipant />") in order to receive
+   *        only a certain mesh region. Have a look at the website under
+   *        https://precice.org/couple-your-code-direct-access.html or
+   *        navigate manually to the page  [sidebar]->Couple your code
+   *        -> Advanced topics -> Accessing received meshes directly for
+   *        a comprehensive documentation
+   *
+   * @experimental
+   *
+   * This function is required if you don't want to use the mapping
+   * schemes in preCICE, but rather want to use your own solver for
+   * data mapping. As opposed to the usual preCICE mapping, only a
+   * single mesh (from the other participant) is now involved in this
+   * situation since an 'own' mesh defined by the participant itself
+   * is not required any more. In order to re-partition the received
+   * mesh, the participant needs to define the mesh region it wants
+   * read data from and write data to.
+   * Defining a bounding box for serial runs of the solver (not to
+   * be confused with serial coupling mode) is valid. However, a
+   * warning is raised in case vertices are filtered out completely
+   * on the receiving side, since the associated data values of the
+   * filtered vertices are filled with zero data.
+   * The mesh region is specified through a collection of axis-aligned
+   * bounding boxes given by the lower and upper [min and max]
+   * bounding-box limits in each space dimension [x, y, z].
+   *
+   * @param[in] meshID ID of the mesh you want to access through the bounding box
+   * @param[in] boundingBoxCollection collection of (axis aligned) bounding boxes which
+   *            has in 3D the format
+   *            [x_min0, x_max0, y_min0, y_max0, z_min0, z_max0, x_min1 ... ]
+   * @param[in] size number of bounding boxes.
+   *
+   * @warning This functions is currently only implemented for nBoundingBoxes = 1.
+   *
+   * @pre 'initialize' has not yet been called.
+   */
+  void setBoundingBoxes(
```

Comment:
I don't see any confusion for users as well. In particular, as we mean the same kind of bounding boxes, we also use in the partitioning. We use these bounding boxes for the re-partitioning.

Still, I like some of the suggestions. `setMeshAccessRegion` sounds good (better than `setBoundingBox`) if we only allow a single call of the function. 

I don't like "mark" as it would indicate that sth is already there and we just tag it. But here nothing is there yet.

Reply:
Renamed it now to `setMeshAccessRegion`